ignite/cmd/network.go

```diff
@@ -125,7 +126,16 @@ func (n NetworkBuilder) Network(options ...network.Option) (network.Network, err
 			return network.Network{}, errors.Wrap(err, "make sure that this account exists, use 'ignite account -h' to manage accounts")
 		}
 	}
-	return network.New(*cosmos, account, options...)
+
+	options = append(options,
+		network.CollectEvents(n.ev),
+		network.WithCampaignQueryClient(campaigntypes.NewQueryClient(cosmos.Context())),
```

Comment:
Can we remove these lines `WithCampaign` and set them by default [here](https://github.com/ignite-hq/cli/blob/4069765/ignite/services/network/network.go#L91-L94)? 

And in the tests, we'll be overwriting them with the mock ones through Options.

Reply:
In order to create a client we need a connection instance(`cosmos.Context()` in our case) If we move client initiation inside of the constructor we will rely that cosmos object which is passed in constructor has correct `.Context()`. But in case of testing we are passing mocked `cosmos` and it doesn't have Context(). Mocked `cosmos` will panic on `Context()` call because it was not previously defined

Reply:
I don't feel like this is relevant. Can you try to move these lines (actual implementation) there and we see if it'll fail or not? We don't touch the ones in the test suite, only this one.

This way, by default network.New will initialize with a real implementation and in the tests we'll be setting the mocked ones via WithX options.

Reply:
Tests are using `network.New()` and they cannot provide connection for defaults initialization.

Reply:
Moved defaults definition inside of the constructor and add behavior for 'Context()' function on cosmos mock as we discussed on the call.